### PR TITLE
Sets up a basic CLI interface

### PR DESCRIPTION
**Pull request by stuartlynn**
- Installs clap 
- Defines a new module for the CLI
- Implements some placeholder commands with Arg structs 

TODO 

- [x]  Figure out validation for things like bounding boxes 
- [x] Implement a struct to represent Data Requests more generally and implement a From trait for it from the CLI parser. This will let us potentially reuse that same struct for a config file version of invoking the app.
- [x] Figure out an ergonomic way to map commands to executors. 
- - Just a big match statement? 
- - Trait and impl on the Args object for each Command with dynamic dispatch? 
- - enum dispatch like pattern on the Commands object?

### CODE REVIEW

**Comment by stuartlynn** [comment]
Last push added a pattern for running command more ergonomically. 

Now we declare a trait RunCommand:

```rust
#[enum_dispatch]
pub trait RunCommand{
    fn run(&self)->Result<()>;
} 
```

which each Command needs to implement. We then just invoke command.run() in the main.rs entrypoint  to run the command. For example 

```rust
impl RunCommand for DataCommand{
    fn run(&self)->Result<()> {
        println!("Running Data Command");
        println!("{self:#?}");
        Ok(())
    }
}
```

In addition I stubbed out a lib struct to start to represent a data request. We have implemented `From<DataCommand> for DataRequestSpec`. This will probably grow and change in time but is a good starting point for now.

```rust
impl From<DataCommand> for DataRequestSpec{
        fn from(value: DataCommand) -> Self {
            let region = if let Some(bbox) = value.bbox{
                vec![RegionSpec::BoundingBox(bbox)] 
            }
            else{
                vec![]
            };
            let metrics = vec![];
            DataRequestSpec{
                region,
                metrics
            }

    }
}
```

File: src/lib.rs [hunk 1]
```diff
@@ -1 +1,16 @@
+use anyhow::Result;
+use metadata::{load_metadata, MetricMetadata, SourceDataRelease};
+
+pub mod data_request_spec;
 pub mod metadata;
+
+pub struct Popgetter {
+    metadata: SourceDataRelease,
```
Comment (yongrenjie) [inline src/lib.rs:8]:
```suggestion
    pub metadata: SourceDataRelease,
```

File: src/lib.rs [hunk 2]
```diff
@@ -1 +1,16 @@
...
+}
+
+impl Popgetter {
+    fn new() -> Result<Self> {
```
Comment (yongrenjie) [inline src/lib.rs:12]:
```suggestion
    pub fn new() -> Result<Self> {
```

**Review by yongrenjie — COMMENTED** [review]
Looks good. Just proposing changes to silence the other two warnings.

I feel like the spurious FromStr warning should be reported upstream if it isn't already

**Comment by yongrenjie** [comment]
Macros are too magic :)

**Comment by yongrenjie** [comment]
I think it's this issue here: https://github.com/rust-lang/rust-analyzer/issues/15501

**Pull request merged**